packages/lib/views/pagination/pagination.test.js

```diff
@@ -30,194 +31,231 @@ describe('pagination macro', () => {
 			'Page 1 link should be present'
 		);
-		assert.ok(
-			rendered.includes(
-				'<a class="govuk-link govuk-pagination__link" href="?page=2" aria-label="Page 2" aria-current="page">'
-			),
+		// page 2 as current page
+		assert.match(
+			rendered,
+			/<a[^>]*href="[^"]*page=2"[^>]*aria-label="Page 2"[^>]*aria-current="page"[^>]*>[\s\S]*2[\s\S]*<\/a>/,
 			'Page 2 link should be current page and present'
 		);
-		assert.ok(
-			rendered.includes('<a class="govuk-link govuk-pagination__link" href="?page=3" aria-label="Page 3">'),
+		// page 3 as a link
+		assert.match(
+			rendered,
+			/<a[^>]*href="[^"]*page=3"[^>]*aria-label="Page 3"[^>]*>[\s\S]*1[\s\S]*<\/a>/,
 			'Page 3 link should be present'
 		);
 
+		// Elipses not present
 		assert.ok(
-			!rendered.includes('<li class="govuk-pagination__item govuk-pagination__item--ellipsis">'),
+			!/<li[^>]*class="govuk-pagination__item govuk-pagination__item--ellipsis"[^>]*>/.test(rendered),
 			'Elipses should not be present'
 		);
 
-		assert.ok(
-			rendered.includes('<a class="govuk-link govuk-pagination__link" href="?page=1" rel="prev">'),
-			'Previous page link should be rendered'
-		);
-		assert.ok(
-			rendered.includes('<a class="govuk-link govuk-pagination__link" href="?page=3" rel="next">'),
-			'Next page link should be rendered'
-		);
+		// Previous page link
+		assert.match(rendered, /<a[^>]*href="[^"]*page=1"[^>]*rel="prev"[^>]*>/, 'Previous page link should be rendered');
+
+		// Next page link
+		assert.match(rendered, /<a[^>]*href="[^"]*page=3"[^>]*rel="next"[^>]*>/, 'Next page link should be rendered');
 	});
 
 	it('should render pagination correctly and append page url parameter with ampersand', () => {
 		const currentPage = 2;
 		const totalPages = 3;
-		const currentUrl = `/applications/${applicationId}/written-representations?itemsPerPage=25&page=2`;
+		const baseUrl = `/applications/${applicationId}/written-representations`;
+		const queryParams = { itemsPerPage: 25, page: 2 };
 
 		const rendered = nunjucks.renderString(wrapperTemplate, {
 			currentPage,
 			totalPages,
-			currentUrl
+			baseUrl,
+			queryParams
 		});
 
-		assert.ok(
-			rendered.includes(
-				'<a class="govuk-link govuk-pagination__link" href="/applications/cfe3dc29-1f63-45e6-81dd-da8183842bf8/written-representations?itemsPerPage=25&amp;page=1" aria-label="Page 1">'
-			),
+		// Page 1 as a link
+		assert.match(
+			rendered,
+			/<a[^>]*href="[^"]*itemsPerPage=25[^"]*page=1"[^>]*aria-label="Page 1"[^>]*>[\s\S]*1[\s\S]*<\/a>/,
 			'Page 1 link should be present'
 		);
-		assert.ok(
-			rendered.includes(
-				'<a class="govuk-link govuk-pagination__link" href="/applications/cfe3dc29-1f63-45e6-81dd-da8183842bf8/written-representations?itemsPerPage=25&amp;page=2" aria-label="Page 2" aria-current="page">'
-			),
+
+		// Page 2 as current page
+		assert.match(
+			rendered,
+			/<a[^>]*href="[^"]*itemsPerPage=25[^"]*page=2"[^>]*aria-label="Page 2"[^>]*aria-current="page"[^>]*>[\s\S]*2[\sS]*<\/a>/,
```

Comment:
I think this might be a typo if you are checking for whitespaces and new lines
```suggestion
			/<a[^>]*href="[^"]*itemsPerPage=25[^"]*page=2"[^>]*aria-label="Page 2"[^>]*aria-current="page"[^>]*>[\s\S]*2[\s\S]*<\/a>/,
```